src/main/java/seedu/address/logic/commands/DeleteEventCommand.java

```diff
@@ -0,0 +1,73 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.event.Event;
+
+/**
+ * Deletes an event identified using it's displayed index from the address book.
+ */
+public class DeleteEventCommand extends Command {
+
+    public static final String COMMAND_WORD = "delev";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the event identified by the index number used in the displayed event list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + "1";
+
+    public static final String MESSAGE_DELETE_EVENT_SUCCESS = "Deleted Event: %1$s";
+
+    public final Index targetIndex;
+
+    public DeleteEventCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Event> lastShownList = model.getFilteredEventList();
+
+        if (model.isAnEventSelected()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_DELETE_EVENT);
```

Comment:
The command should only not work when delev is trying to delete the event that it is currently selected, in other words
it is okay to delete events that are not selected.

So you should make a method in modelmanager.java to check whether the event that it is trying to delete is the event that is currently selected. Sorry if my phrasing was vague before.